### ISSUE TITLE
Statistics not properly updated if accessed before grounding

### ISSUE THREAD

**Issue by rkaminsk** [potassco/clingo#81]
```python
import clingo

ctl = clingo.Control([])
ctl.add("base",[],"""
a.
""")

#print(ctl.statistics)
ctl.ground([("base",[])])
ctl.solve()
print(ctl.statistics)
```

A large part of the statistics is missing in the second step if the first call is uncommented.

**Comment by rkaminsk** [comment]
@BenKaufmann 

If `ClaspFacade::getStats()` is used to access the statistics before solving, then the "lp" and "lpStep" entries do not appear in the statistics. I think the reason is that they are only initialized in the `ClingoView` constructor. When the `lpStats` are not yet available at this point, they won't appear later when the become available.

Any ideas here? Right now the only condition on `getStats` is `!solving`.

**Comment by rkaminsk** [comment]
@BenKaufmann  The following patch seems to fix the problem. I just don't know if it is the best way to fix the issue. Also the corresponding code from the ClingoView constructor could be removed if this is done in update.

```patch
diff --git a/src/clasp_facade.cpp b/src/clasp_facade.cpp
index c99ed7d..c46bd6f 100644
--- a/src/clasp_facade.cpp
+++ b/src/clasp_facade.cpp
@@ -737,6 +737,11 @@ Potassco::AbstractStatistics::Key_t ClaspFacade::Statistics::ClingoView::user()
        return user_;
 }
 void ClaspFacade::Statistics::ClingoView::update(const ClaspFacade::Statistics& stats) {
+       const ClaspFacade& f = *stats.self_;
+       if (f.step_.lpStats() && !problem_.find("lp")) {
+               problem_.add("lp", StatisticObject::map(f.step_.lpStats()));
+               if (f.incremental()) { problem_.add("lpStep", StatisticObject::map(f.step_.lpStep())); }
+       }
        StatsCallbacks sources = stats.external();
        if (!Potassco::empty(sources)) {
                user_ = add(root(), "user_defined", Potassco::Statistics_t::Map);
```

**Comment by BenKaufmann** [comment]
@rkaminsk 
`ClaspFacade::getStats()` was never meant to be called before having any meaningful statistics, i.e. before solving ended. Originally, the function had a stricter precondition, namley that the step was `solved()`. However, in an endeavour to support poorly written python scripts and to allow getting statistics of the previous step until the next step starts, I weakened the precondition. While I'm strongly against supporting the usage you propose, it seems that now that I messed up the precondition, I'm bound to support it :(

I'll fix this together with the external stats stuff.
